src/Data/ListsStorage.js

```
@@ -0,0 +1,80 @@
+import IdGenerator from "../UI/IdGenerator/idGenerator";
+
+class ListsStorage {
+    /**
+     * Take care about datas in LocalStorage
+     * @param {String} listName 
+     * @param {JSON} lists
+     */
+    constructor(listName) {
+        let lists = localStorage.getItem("lists");
+        
+        if (listName === undefined) {
+            return;
+        } else if (lists) {
+            this.addListToStorage(listName);
+        } else {
+            this.addStorage(listName);
+        }
+    }
+    
+    addStorage(listName) {
+        this.lists = [
+            {
+                listId: IdGenerator(),
+                listName: listName,
+                tasks: []
+            }
+        ];
+        localStorage.setItem("lists", JSON.stringify(this.lists));
+    }   
+
+    addListToStorage(listName) {
+        let actualLists = JSON.parse(localStorage.getItem("lists"));
+        actualLists.push({
+            listId: IdGenerator(),
+            listName: listName,
+            tasks: []
+        });
+        localStorage.setItem("lists", JSON.stringify(actualLists));
+    }   
+
+    addNewTaskToStorage(listId, taskName) {
+        let actualLists = JSON.parse(localStorage.getItem("lists"));
+        let choosenList = actualLists.filter(list => list.listId === listId);
+        choosenList[0].tasks.push(
```

There's no guarantee element will be found, element at 0 or found element can be undefined. We should decide what to do, report error to UI, that silently exception, or silently skip.

---

changed